wallet-api/wallet_rpc.json

```
@@ -0,0 +1,677 @@
+          },
+          "contract_class": {
+            "title": "Contract class",
+            "description": "The class to be declared",
+            "$ref": "#/components/schemas/CONTRACT_CLASS"
+          }
+        },
+        "required": ["compiled_class_hash", "contract_class"]
+      },
+      "STARKNET_CHAIN": {
+        "title": "Starknet Chain",
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string"
+          },
+          "chain_id": {
+            "title": "Starknet Chain Id",
+            "$ref": "#/components/schemas/CHAIN_ID"
+          },
+          "chain_name": {
+            "type": "string"
+          },
+          "rpc_urls": {
+            "type": "array",
+            "items": {
+              "type": "string",
+              "format": "uri"
+            },
+            "minItems": 1,
+            "uniqueItems": true
+          },
+          "block_explorer_url": {
+            "type": "array",
+            "items": {
+              "type": "string",
+              "format": "uri"
+            },
+            "minItems": 1,
+            "uniqueItems": true
+          },
+          "native_currency": {
+            "title": "Native Currency",
+            "$ref": "#/components/schemas/ASSET"
+          },
+          "icon_urls": {
+            "type": "array",
+            "items": {
+              "type": "string",
+              "format": "uri"
+            },
+            "minItems": 1,
+            "uniqueItems": true
+          }
+        },
+        "required": ["id", "chain_id", "chain_name"]
+      },
+      "STARKNET_MERKLE_TYPE": {
+        "title": "Starknet Merkle Type",
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "type": {
+            "type": "string",
+            "enum": ["merkletree"]
+          },
+          "contains": {
+            "type": "string"
+          }
+        },
+        "required": ["name", "type", "contains"]
+      },
+      "STARKNET_TYPE": {
+        "title": "Starknet Type",
+        "description": "A single type, as part of a struct. The `type` field can be any of the EIP-712 supported types",
+        "schema": {
+          "anyOf": [
+            {
+              "type": "object",
+              "properties": {
+                "name": { "type": "string" },
+                "type": { "type": "string" }
+              },
+              "required": ["name", "type"]
+            },
+            {
+              "$ref": "#/components/schemas/STARKNET_MERKLE_TYPE"
+            }
+          ]
+        }
+      },
+      "TYPED_DATA": {
+        "title": "Typed Data",
+        "type": "object",
+        "properties": {
+          "types": {
+            "description": "Defines the types of the typed data object",
+            "type": "array",
+            "itmes": {
+              "$ref": "#/components/schemas/STARKNET_TYPE"
+            }
+          },
```

`types` should be an object with arbitrary keys where each key maps to a `StarknetType` array ([SNIP-12 example](https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-12.md#json-example)). 

Not sure if this suggestion code is a valid way to define it (I can't find`additionalProperties` on `open-rpc.org` but it can be found on `json-schema.org` and is used in [one other place](https://github.com/starkware-libs/starknet-specs/blob/v0.7.1/api/starknet_api_openrpc.json#L1279) in the spec):
```suggestion
          "types": {
            "type": "object",
            "additionalProperties": {
              "type": "array",
              "items": {
                "$ref": "#/components/schemas/STARKNET_TYPE"
              }
            }
          },
```

---

Thanks @penovicp! applied this and your suggestions for SNIP 12